mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1230,21 +1232,24 @@ impl TakerSwap {
             },
         };
         drop(abort_send_handle);
+
         let mut swap_events = vec![];
-        if let Some(instructions) = payload.instructions() {
-            match self.taker_coin.validate_taker_payment_instructions(
+        let instructions = match payload.instructions() {
+            Some(instructions) => match self.taker_coin.validate_taker_payment_instructions(
                 instructions,
                 &self.r().secret_hash.0,
                 self.taker_amount.clone().into(),
             ) {
-                Ok(instructions) => swap_events.push(TakerSwapEvent::TakerPaymentInstructionsReceived(instructions)),
+                Ok(instructions) => Some(instructions),
                 Err(e) => {
                     return Ok((Some(TakerSwapCommand::Finish), vec![
                         TakerSwapEvent::MakerPaymentValidateFailed(e.to_string().into()),
-                    ]));
+                    ]))
                 },
-            }
-        }
+            },
+            None => None,
+        };
+        swap_events.push(TakerSwapEvent::TakerPaymentInstructionsReceived(instructions));
```

Comment:
Could you please clarify why do we need to push `TakerSwapEvent::TakerPaymentInstructionsReceived` even if the `instructions` is None?

Reply:
I did this for uniformity; to have the same swap events for all kind of swaps. I can revert this if you think it would be better, it will require some changes in unit tests by passing success, error events to `wait_for_swaps_finish_and_check_status` https://github.com/KomodoPlatform/atomicDEX-API/blob/3279aa6a6cbb4c5347be27bfd52032e13773e79a/mm2src/mm2_test_helpers/src/for_tests.rs#L2539